File: app/src/main/java/com/cookplan/recipe/import_recipe/approve_result/ImportRecipePresenter.java
package com.cookplan.recipe.import_recipe.approve_result;

import com.cookplan.models.Ingredient;
import com.cookplan.models.MeasureUnit;
import com.cookplan.models.Product;
import com.cookplan.models.ProductCategory;
import com.cookplan.models.Recipe;

import java.util.List;

/**
 * Created by DariaEfimova on 08.06.17.
 */

public interface ImportRecipePresenter {

    void importRecipeFromUrl(String uri);

    List<Product> getAllProductsList();

    void saveRecipe(Recipe recipe);

    void saveProductAndIngredient(String key, ProductCategory category, String name, double amount, MeasureUnit measureUnit);

    void saveIngredient(String key, Ingredient ingredient);
}

File: app/src/main/java/com/cookplan/recipe/import_recipe/approve_result/ImportRecipePresenter.kt
package com.cookplan.recipe.import_recipe.approve_result

import com.cookplan.models.Ingredient
import com.cookplan.models.MeasureUnit
import com.cookplan.models.Product
import com.cookplan.models.ProductCategory
import com.cookplan.models.Recipe

/**
 * Created by DariaEfimova on 08.06.17.
 */

interface ImportRecipePresenter {

    fun getAllProductsList(): List<Product>

    fun importRecipeFromUrl(uri: String)

    fun saveRecipe(recipe: Recipe)

    fun saveProductAndIngredient(key: String, category: ProductCategory, name: String, amount: Double, measureUnit: MeasureUnit)

    fun saveIngredient(key: String, ingredient: Ingredient)
}
